hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/search/search.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Input, Select } from 'antd';
+import { Option } from '@/v2/components/select/singleSelect';
+
+type SearchProps = {
+  searchColumn?: string;
+  searchOptions?: Option[];
+  onSearch: (
+    arg0: string,
+    arg1: React.ChangeEvent<HTMLInputElement>
+    | React.MouseEvent<HTMLElement, MouseEvent>
+    | React.KeyboardEvent<HTMLInputElement>
+    | undefined
+  ) => void;
+  onChange: (
+    value: string,
+    //OptionType, OptionGroupData and OptionData are not
+    //currently exported by AntD hence set to any
+    option: any
+  ) => void;
+}
+
+const Search: React.FC<SearchProps> = ({
+  searchColumn,
+  searchOptions,
+  onSearch = () => {},
+  onChange = () => {}
```

Comment:
While you've set default values for `onSearch` and `onChange`, it might be more explicit and clear to set them using default props. This can make the code easier to understand, especially for developers who might not be familiar with default parameters in function signatures.

Reply:
When using defaultProps actually I tested with React v18 as well to future-proof the code.
I get the following warning as React will deprecate the defaultProps in future releases.
<img width="552" alt="Screenshot 2024-08-13 at 12 33 16" src="https://github.com/user-attachments/assets/98a28fa6-ddf4-42b3-9c5c-fef97b46ca08">

Reference proposal for the same is [here](https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md)
I believe it would be better to write the newer syntax so that we can seamlessly upgrade the libraries in the future

Reply:
Added the comments for the functions for other contributors to better understand